### PR TITLE
Add -basedir option to specify base user directory 

### PR DESCRIPTION
**Pull request by mwpenny**
## Changes:

Added a command-line argument to specify the location of the base user directory. This is useful for maintaining multiple save files or for debugging.

## Legal Stuff:

By submitting this pull request, I confirm that...

- [x] My changes may be used in a future commercial release of VVVVVV (for
  example, a 2.3 update on Steam for Windows/macOS/Linux)
- [x] I will be credited in a `CONTRIBUTORS` file for all of said releases, but
  will NOT be compensated for these changes

### CODE REVIEW

File: desktop_version/CONTRIBUTORS.txt [hunk 1]
```diff
@@ -3,13 +3,15 @@ Contributors
 
 (Ordered alphabetically by last name.)
 
+* Matt "Stelpjo" Aaldenberg
```
Comment (mwpenny) [inline desktop_version/CONTRIBUTORS.txt]: Re-ordered to be alphabetical

File: desktop_version/src/Game.cpp [hunk 2]
```diff
@@ -468,12 +468,15 @@ void Game::init(void)
     patrons.push_back("Timothy Bragan");
 
     /* CONTRIBUTORS.txt, again listed alphabetically (according to `sort`) by last name */
+    githubfriends.push_back("Matt \"Stelpjo\" Aaldenberg");
```
Comment (mwpenny) [inline desktop_version/src/Game.cpp]: Some names in CONTRIBUTORS.txt were missing from `githubfriends`

File: desktop_version/src/FileSystemUtils.cpp [hunk 3]
```diff
@@ -32,14 +32,20 @@ int mkdir(char* path, int mode)
 #define MAX_PATH PATH_MAX
 #endif
 
+#ifdef _WIN32
```
Comment (flibitijibibo) [inline desktop_version/src/FileSystemUtils.cpp]: PhysFS provides this in their API, can you use this instead?

Comment (mwpenny) [reply desktop_version/src/FileSystemUtils.cpp]: Okay, good to know. I've changed it to use the existing function

**Pull request merged**